src/handlers/uniswapLiquidity.ts

```diff
@@ -14,130 +15,303 @@ import { UNISWAP_V1_EXCHANGE_ABI } from '../references/uniswap';
 import { web3Provider } from './web3';
 import logger from 'logger';
 
+interface LiquidityToken {
+  address: string;
+  balance: string;
+  name: string;
+  symbol: string;
+}
+
 export interface LiquidityInfo {
+  address: string;
   balance: string;
-  ethBalance: string;
-  ethReserve: BigNumber;
-  token: { balance: string; decimals: number; name: string; symbol: string };
-  tokenAddress: string;
+  price?: {
+    changed_at?: number;
+    value?: number;
+    relative_change_24h?: number;
+  };
+  tokens: LiquidityToken[];
   totalSupply: string;
   uniqueId: string;
 }
 
+export interface LPToken {
+  asset: {
+    asset_code: string;
+    price?: {
+      changed_at?: number;
+      value?: number;
+      relative_change_24h?: number;
+    };
+    type: string;
+  };
+  quantity: string;
+}
+
+const getAssetCode = (token: LPToken): string => get(token, 'asset.asset_code');
+
+const getTokenDetails = async (
+  chainId: ChainId,
+  tokenAddress: string,
+  pairs: Record<string, SwapCurrency>
+) => {
```

Comment:
You might wanna create a new type TokenDetails and set it as the return type here.